sqlglot/dialects/duckdb.py

```diff
@@ -96,6 +97,21 @@ def _json_format_sql(self: generator.Generator, expression: exp.JSONFormat) -> s
     return f"CAST({sql} AS TEXT)"
 
 
+def _ensure_utf8(charset: exp.Literal) -> None:
+    if charset.name.lower() != "utf-8":
+        raise UnsupportedError(f"Unsupported charset {charset}")
+
+
+def _decode_sql(self: generator.Generator, expression: exp.Decode) -> str:
+    _ensure_utf8(expression.args["charset"])
```

Comment:
charset may not exist since you made it false, so use get

Reply:
Now that I have refactored the parser, I think that charset is always available.